core/src/main/java/hudson/model/FileFingerprintStorage.java

```diff
@@ -0,0 +1,90 @@
+package hudson.model;
+
+import com.thoughtworks.xstream.converters.basic.DateConverter;
+import hudson.BulkChange;
+import hudson.Util;
+import hudson.model.listeners.SaveableListener;
+import hudson.util.AtomicFileWriter;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class FileFingerprintStorage {
```

Comment:
Unless there are startup sequence related issues, this should be fairly straightforward to do in a "Jenkins" way:

* Remove the field in `Fingerprint`.
* Annotate this class `@Extension(ordinal=100)` after making sure it has a no-arg constructor.
* Add an abstract superclass `FingerprintStorage implements ExtensionPoint` and extend it here.
* Have a `static FingerprintStorage get()` in the superclass that returns `ExtensionList.lookup(FingerprintStorage.class).get(0)`.

Any `@Extension WhateverFingerprintStorage extends FingerprintStorage` will take precedence.

Even if the final PR looks nothing like I write above, looking into these Jenkins-isms of you don't know then yet will prepare you for writing code "the Jenkins way" 😃

Reply:
Hi @daniel-beck Thanks a lot for the review :) Implementing the Extension point was indeed my next step, but you gave me such good precise and clear direction! Thanks a ton!

Preparing for "the Jenkins way" is what I am here to learn! :)

Reply:
Would also recommend looking into https://javadoc.jenkins.io/jenkins/model/ArtifactManagerConfiguration.html and implementations of https://www.jenkins.io/doc/developer/extensions/jenkins-core/#artifactmanagerfactory -- seems they are fairly similar to what you're doing here.